azure-pipelines.yml

```diff
@@ -149,6 +150,67 @@ jobs:
         trigger_sha: '$(trigger_sha)'
     - template: ci/report-end.yml
 
+- job: split_release
+  dependsOn: [ "check_for_release", "Linux", "macOS", "Windows" ]
+  condition: and(succeeded(),
+                 eq(dependencies.check_for_release.outputs['out.is_release'], 'true'),
+                 eq(dependencies.check_for_release.outputs['out.split_release_process'], 'true')
+                 eq(variables['Build.SourceBranchName'], 'main'))
+  pool:
+    name: 'ubuntu_20_04'
+    demands: assignment -equals default
+  variables:
+    release_sha: $[ dependencies.check_for_release.outputs['out.release_sha'] ]
+    release_tag: $[ dependencies.check_for_release.outputs['out.release_tag'] ]
+    trigger_sha: $[ dependencies.check_for_release.outputs['out.trigger_sha'] ]
+  steps:
+    - checkout: self
+      persistCredentials: true
+    - bash: |
+        set -euo pipefail
+        git checkout $(release_sha)
+      name: checkout_release
+    - template: ci/bash-lib.yml
+      parameters:
+        var_name: bash-lib
+    - bash: |
+        set -euo pipefail
+        source $(bash-lib)
+        if git tag v$(release_tag) $(release_sha); then
+          git push origin v$(release_tag)
+        fi
+    - task: DownloadPipelineArtifact@0
+      inputs:
+        artifactName: linux-release
+        targetPath: $(Build.StagingDirectory)/release-artifacts
+      condition: and(succeeded(), not(eq(variables['skip-github'], 'TRUE')))
+    - task: DownloadPipelineArtifact@0
+      inputs:
+        artifactName: macos-release
+        targetPath: $(Build.StagingDirectory)/release-artifacts
+      condition: and(succeeded(), not(eq(variables['skip-github'], 'TRUE')))
+    - task: DownloadPipelineArtifact@0
+      inputs:
+        artifactName: windows-release
+        targetPath: $(Build.StagingDirectory)/release-artifacts
+      condition: and(succeeded(), not(eq(variables['skip-github'], 'TRUE')))
+    - template: ci/tell-slack-failed.yml
```

Comment:
You probably want to move this to the end, otherwise it will only report failures before this point.

Reply:
9607a887a681828e056613f9c889f6c68e88c407